build.sc

```diff
@@ -87,71 +82,20 @@ object v {
+    s"-Wconf:${scala2WarnConf.mkString(",")}"
   )
 }
 
-object utils extends Module {
-
-  val architecture = System.getProperty("os.arch")
-  val operationSystem = System.getProperty("os.name")
-
-  val mac = operationSystem.toLowerCase.startsWith("mac")
-  val linux = operationSystem.toLowerCase.startsWith("linux")
-  val windows = operationSystem.toLowerCase.startsWith("win")
-  val amd64 = architecture.matches("^(x8664|amd64|ia32e|em64t|x64|x86_64)$")
-  val aarch64 = architecture.equals("aarch64") | architecture.startsWith("armv8")
-
+private object utils extends Module {
   val firtoolVersion = {
     val j = _root_.upickle.default.read[Map[String, String]](os.read(millSourcePath / os.up / "etc" / "circt.json"))
     j("version").stripPrefix("firtool-")
   }
 
-  // use T.persistent to avoid download repeatedly
-  def circtInstallDir: T[os.Path] = T.persistent {
-    T.ctx().env.get("CIRCT_INSTALL_PATH") match {
-      case Some(dir) => os.Path(dir)
-      case None =>
-        T.ctx().log.info("Use CIRCT_INSTALL_PATH to vendor circt")
-        val tarPath = T.dest / "circt.tar.gz"
-        if (!os.exists(tarPath)) {
-          val url = v.circt(
-            firtoolVersion,
-            if (linux) "linux" else if (mac) "macos" else throw new Exception("unsupported os"),
-            // circt does not yet publish for macos-aarch64, use x64 for now
-            if (amd64 || mac) "x64" else throw new Exception("unsupported arch")
-          )
-          T.ctx().log.info(s"Downloading circt from ${url}")
-          mill.util.Util.download(url, os.rel / "circt.tar.gz")
-          T.ctx().log.info(s"Download Successfully")
-        }
-        os.proc("tar", "xvf", tarPath, "--strip-components=1").call(T.dest)
-        T.dest
-    }
-  }
-
-  // use T.persistent to avoid download repeatedly
-  def jextractInstallDir: T[os.Path] = T.persistent {
-    T.ctx().env.get("JEXTRACT_INSTALL_PATH") match {
-      case Some(dir) => os.Path(dir)
-      case None =>
-        T.ctx().log.info("Use JEXTRACT_INSTALL_PATH to vendor jextract")
-        val tarPath = T.dest / "jextract.tar.gz"
-        if (!os.exists(tarPath)) {
-          val url = v.jextract(
-            21,
-            "1-2",
-            if (linux) "linux" else if (mac) "macos" else throw new Exception("unsupported os"),
-            // There is no macos-aarch64 for jextract 21, use x64 for now
-            if (amd64 || mac) "x64" else if (aarch64) "aarch64" else throw new Exception("unsupported arch")
-          )
-          T.ctx().log.info(s"Downloading jextract from ${url}")
-          mill.util.Util.download(url, os.rel / "jextract.tar.gz")
-          T.ctx().log.info(s"Download Successfully")
-        }
-        os.proc("tar", "xvf", tarPath, "--strip-components=1").call(T.dest)
-        T.dest
-    }
+  def isScala3(ver: String): Boolean = ver match {
+    case "3.4.2" => true
+    case "2.13.15" => false
+    case _ => false // todo throw error here?
   }
```

Comment:
```suggestion
  def isScala3(ver: String): Boolean = ver.startsWith("3.")
```

From looking at other examples (e.g. scala-cli's build.sc), it's more common to just do `crossScalaVersion.startsWith("3.")`

I do think it's important that you use `crossScalaVersion` instead of `crossValue` because the former is guaranteed to always be a Scala version while the latter can be whatever we want it to be (it just happens that we're using full Scala versions right now).